common/src/update.rs

```diff
@@ -328,6 +328,19 @@ pub fn hex_schema<const N: usize>(gen: &mut SchemaGenerator) -> Schema {
     schema.into()
 }
 
+fn artifact_hash_schema(gen: &mut SchemaGenerator) -> Schema {
+    let mut schema: SchemaObject = hex_schema::<32>(gen).into();
+    schema.extensions.insert(
+        "x-rust-type".into(),
+        serde_json::json!({
+            "crate": env!("CARGO_PKG_NAME"),
+            "version": env!("CARGO_PKG_VERSION"),
```

Comment:
This doesn't seem quite right in that the version is intended to fixed at the version at which the type became public... which isn't particularly relevant here. Given the nature of the monorepo, I'd suggest

```suggestion
            "version": "*",
```